libqtile/widget/camcheck.py

```diff
@@ -0,0 +1,43 @@
+# A qtile widget written to show active indication for mic or camera
+# written by bhauvk sharma (https://github.com/desdemona2)
+
+from libqtile.widget import base
+import os
+
+
+class active(base.ThreadPoolText):
+    """widget to show indication if
+    camera and mic are active in status bar"""
+    
+    defaults = [
+           (
+            "update_interval", 
+            1, 
+            "Update interval in seconds, if none, the "
+            "widget updates whenever it's done'."
+            ),
+    ]
+
+
+    def __init__(self, **config):
+        super().__init__("", **config)
+        self.add_defaults(active.defaults)
+
+
+    def poll(self):
+        camera = os.system("fuser /dev/video0")
+        mic = os.system("fuser /dev/snd/pcmC0D0c")
+    
+        def mictest():
+            if mic == 256:
+                return ""
+            else:
+                return "📢"
+        
+        def camtest():
+            if camera == 256:
+                return ""
+            else:
+                return "📸"
+
+        return f"{mictest()} {camtest()}"
```

Comment:
Format should be configurable too e.g.:
``` python
# defaults can include...
defaults = [
    ...
    fmt = "{mic} {cam}"
    ...
]

...

def poll(self):
    camera = os.system(f"fuser {self.camera}")
    mic = os.system(f"fuser {self.microphone}")
    vals["cam"] = "" if cam == 256 else "📸"
    vals["mic"] = "" if mic == 256 else "📢"

    return self.fmt.format(**vals)

Reply:
`from libqtile.widget import base
import os


class active(base.ThreadPoolText):
    """widget to show indication if
    camera and mic are active in status bar"""

    defaults = [
        (
            "update_interval",
            1,
            "Update interval in seconds, if none, the "
            "widget updates whenever it's done'.",
        ),
        # ("fmt", "{mic_str} {cam_str}", "Display format for output"),
    ]

    def __init__(
        self, camDev="video0", sndDev="pcmC0D0c", camAct="📸", micAct="📢", **config
    ):
        super().__init__("", **config)
        self.add_defaults(active.defaults)

        self.camDev = camDev
        self.micDev = sndDev

        self.camAct = camAct
        self.micAct = micAct

    def poll(self):

        mic = os.system(f"fuser /dev/snd/{self.micDev}")
        camera = os.system(f"fuser /dev/{self.camDev}")

        mic_str = "" if mic == 256 else self.micAct
        cam_str = "" if camera == 256 else self.camAct

        return f"{mic_str}{cam_str}"
`
I tried most of the things u said. I am still a student and not a good programmer. You taught me alot of things but i still can't figure out how this formatting thing work in qtile so i am unable to correct the last point you said.